scripts/lint/markdown-heuristic-lint.py

```
@@ -0,0 +1,320 @@
+
+    # Scan each code block line for non-recursive globs.
+    # Use line.strip() for fence detection to handle indented fences correctly.
+    in_code_block = False
+    fence_char: Optional[str] = None
+    current_block_start = 0
+
+    for i, line in enumerate(lines):
+        lstripped = line.strip()
+        stripped = line.rstrip()
+        if not in_code_block:
+            if lstripped.startswith("```") or lstripped.startswith("~~~"):
```

Ruff (0.15.10)

[warning] Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)